### PR TITLE
Ownership & CPPI refactor

### PR DESCRIPTION
**Pull request by awilliamson**
Added lots of ownership, 4 methods + CPPI directly to spawning entities and holograms. Took this chance to refactor MakeSF which was duplicated and on the toolguns, pulled it into SF Lib.

Removed some outdated ownership that was never being used from the hologram entity itself.

### CODE REVIEW

**Comment by Jazzelhawk** [comment]
Model's don't set their posisiton correctly. [L638](https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L638) you are setting the position based on the model's size but it doesn't exist yet. Set the position after spawning the entity or possibly after setting the model.

[This](https://github.com/awilliamson/Starfall/blob/CPPI/lua/weapons/gmod_tool/stools/starfall_processor.lua#L18-L34) and [this](https://github.com/awilliamson/Starfall/blob/CPPI/lua/weapons/gmod_tool/stools/starfall_screen.lua#L57-L73) are not needed anymore (make sure to remove the `local MakeSF` at the top of each file)

Couldn't the `error` be `SF.Throw` on [L604](https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L604)?

Weird indentation on [L709](https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L709) (not Github being weird, it really looks like that)

Missing a space on [L559](https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L559)

**Comment by awilliamson** [comment]
Well shit, just messed that up. I must have rebased ontop of some other changes. Not sure what happened.

The net code indentation is intended. It's to designate a 'block' of net code under that label. I know it's technically wrong, but it's standard convention for GMod stuff.

**Comment by awilliamson** [comment]
@Jazzelhawk Can you check it now? Should all be there, just do a sanity check for me. Had some rebase drama, might have missed something.

Also give it a test for the throw mechanism. 'Should' be ok.

**Comment by Jazzelhawk** [comment]
`SF.throw` actually needs to be `error`, sorry.

Shouldn't the screen `SF.MakeSF` be inside the `RequestCode` block? https://github.com/awilliamson/Starfall/blob/CPPI/lua/weapons/gmod_tool/stools/starfall_screen.lua#L86

When concatenating `typ` it should be `tostring( typ )` in case `typ` isn't a string https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L621

If you look at the net code block in context the block is inside a repeat until statement so needs to be indented once more but the `net.SendToServer` being the same indentation as `net.Start( "starfall_upload" )` https://github.com/awilliamson/Starfall/blob/CPPI/lua/starfall/sflib.lua#L710-L715

**Comment by awilliamson** [comment]
@Jazzelhawk The MakeSF is responsible for the physical spawning. If the code isn't valid, it will just appear red. Users can still then try to upload to it by left click. This is the same mechanism of spawning first as was already in place.

**Comment by awilliamson** [comment]
@Jazzelhawk All fixed now?

**Comment by Xandaros** [comment]
> `SF.throw` actually needs to be `error`, sorry.

Just FYI, there is a call to `error` in `SF.throw`

**Comment by awilliamson** [comment]
@Jazzelhawk Hold off on accepting this, few little tweaks I want to do with the error handling. Adding some notify stuff for users is ply is defined and such.

**Comment by Jazzelhawk** [comment]
@Xandaros `SF.throw` only works in an instance. Since this is to do with making the entity, the instance hasn't been created yet. I did not know this before and therefore thought using `SF.throw` would be fine.

@awilliamson The reason I suggested moving the `SF.MakeSF` inside the `RequestCode` was because this is what is being done in [`starfall_processor.lua`](https://github.com/awilliamson/Starfall/blob/CPPI/lua/weapons/gmod_tool/stools/starfall_processor.lua#L58). Maybe it should be moved inside this file to make it the proper spawning mechanism?

**Comment by Xandaros** [comment]
There is nothing specifically requiring the instance, it is just that there is no code to handle the exception. But yes, it should be `error`, since `throw` is for exceptions that can be caused from within starfall. (Extending this would needlessly add complexity and probably code duplication. A place I don't want to go.)

File: lua/starfall/libs_sv/holograms.lua [hunk 1]
```diff
@@ -318,6 +318,22 @@ function holograms_library.create ( pos, ang, model, scale )
...
         end
 
+		-- Lots of ownership stuff
+		-- Old but still supported officially
+		holoent.owner = instance.player
+
+		-- Questionable use cases
+		-- Based on: https://github.com/garrynewman/garrysmod/blob/master/garrysmod/lua/includes/extensions/entity.lua#L53
+		holoent:SetVar( "Player", instance.player )
+		holoent:SetVar( "Owner", instance.player )
```
Comment (Divran) [inline lua/starfall/libs_sv/holograms.lua:328]: SetVar is the old 2007 deprecated way of doing `ent[index] = value` (or `ent.index = value`) - it's a relic from before entities were tables with metatables.

File: lua/starfall/sflib.lua [hunk 2]
```diff
@@ -580,7 +580,122 @@ if SERVER then
...
+
+		-- NPP / Other PP's hook onto this for ownership usually.
+		ply:AddCleanup( typ, sf )
+
+		ply:AddCount( typ, sf )
+
+		-- Questionable use cases
+		-- Based on: https://github.com/garrynewman/garrysmod/blob/master/garrysmod/lua/includes/extensions/entity.lua#L53
+		sf:SetVar( "Player", ply )
+		sf:SetVar( "Owner", ply )
```
Comment (Divran) [inline lua/starfall/sflib.lua:661]: more deprecated

**Comment by awilliamson** [comment]
It does say "Questionable" above it. Thanks for the heads up, why Garry keeps that junk code in GMod, I've no idea.

**Pull request merged**